### DISCUSSION TITLE
MUC - last message seen by user

### DISCUSSION THREAD

**Discussion by testaccount00075433** [processone/ejabberd#3808] [Questions] [answered]
Hello guys ;) 

I'm pretty new with ejabberd and I have two questions.

1. What is the best approach to check which user seen a message or to see where the user stopped reading? 
2. Do you know any open source clients written in javascript except or similar to https://www.npmjs.com/package/@pazznetwork/ngx-chat?

Thanks and have a great day :) 
Michal

**Comment by weiss** [comment, marked as answer]
> What is the best approach to check which user seen a message or to see where the user stopped reading?

Standard clients typically implement the [XEP-0333: Chat Markers][1] extension for this purpose.

> Do you know any open source clients written in javascript

There's [Converse.js][2] and [JSCX][3], for example.

[1]: https://xmpp.org/extensions/xep-0333.html
[2]: https://conversejs.org
[3]: https://jsxc.org
